File: CSharp/ProjectEuler/ProjectEuler/Problem2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectEuler
{
    class Problem2
    {
        public void run()
        {
            IEnumerable<int> list = generate_list();
            display(list);
        }
        public IEnumerable<int> generate_list()
        {
            int firstArg = 1, secondArg = 1, temp = 0; 
            do{
                temp = secondArg;
                secondArg = fibbonaciSequence(firstArg, secondArg);
                firstArg = temp;
                if(secondArg%2==0){
                    yield return secondArg;
                }
            }while(secondArg<4000000);          
        }
        public int fibbonaciSequence(int firstArg,int secondArg)
        {
            return firstArg+secondArg;
        }
        public void display(IEnumerable<int> list)
        {
            int sum = 0;
            foreach (int element in list)
            {
                sum += element;
            }
            Console.WriteLine("Sum of elements : "+sum);
        }
    }
}

File: Java/ProjectEuler/src/PE/Problem2.java
package PE;

/**
 * Created by Daniel on 2015-08-23.
 */
public class Problem2 {
    public static void main(String[] args){
        int sum = 0;
        int temp = 0;
        int firstValue=1,secondValue = 1;
        do{
           // System.out.println(secondValue);
            temp  = secondValue;
            secondValue=firstValue+secondValue;
            firstValue=temp;
            if(secondValue%2==0) sum+=secondValue;

        }while(secondValue<4000000);
        System.out.println(sum);
    }
}
